src/main/java/com/microsoft/azure/spring/data/documentdb/repository/support/DocumentDbEntityInformation.java

```diff
@@ -122,12 +144,73 @@ private Field getPartitionKeyField(Class<?> domainClass) {
     }
 
     private Integer getRequestUnit(Class<?> domainClass) {
-        Integer ru = 4000;
+        Integer ru = Integer.parseInt(Constants.DEFAULT_REQUEST_UNIT);
         final Document annotation = domainClass.getAnnotation(Document.class);
 
         if (annotation != null && annotation.ru() != null && !annotation.ru().isEmpty()) {
             ru = Integer.parseInt(annotation.ru());
         }
         return ru;
     }
+
+    private Boolean getIndexingPolicyAutomatic(Class<?> domainClass) {
+        Boolean isAutomatic = Boolean.valueOf(Constants.DEFAULT_INDEXINGPOLICY_AUTOMATIC);
+        final DocumentIndexingPolicy annotation = domainClass.getAnnotation(DocumentIndexingPolicy.class);
+
+        if (annotation != null) {
+            isAutomatic = Boolean.valueOf(annotation.automatic());
+        }
+
+        return isAutomatic;
+    }
+
+    private IndexingMode getIndexingPolicyMode(Class<?> domainClass) {
+        IndexingMode mode = Constants.DEFAULT_INDEXINGPOLICY_MODE;
+        final DocumentIndexingPolicy annotation = domainClass.getAnnotation(DocumentIndexingPolicy.class);
+
+        if (annotation != null) {
+            mode = annotation.mode();
+        }
+
+        return mode;
+    }
+
+    private Collection<IncludedPath> getIndexingPolicyIncludePaths(Class<?> domainClass) {
+        final Collection<IncludedPath> pathsCollection = new ArrayList<>();
+        final DocumentIndexingPolicy annotation = domainClass.getAnnotation(DocumentIndexingPolicy.class);
+        final String[] defaultIncludePath = new String[]{Constants.DEFAULT_INCLUDEDPATH};
+        final String[] rawPaths;
+
+        if (annotation == null || annotation.includePaths() == null || annotation.includePaths().length == 0) {
+            rawPaths = defaultIncludePath;
+        } else {
+            rawPaths = annotation.includePaths();
+        }
+
+        for (final String path : rawPaths) {
+            pathsCollection.add(new IncludedPath(path));
+        }
+
+        return pathsCollection;
+    }
+
+    private Collection<ExcludedPath> getIndexingPolicyExcludePaths(Class<?> domainClass) {
+        final Collection<ExcludedPath> pathsCollection = new ArrayList<>();
+        final DocumentIndexingPolicy annotation = domainClass.getAnnotation(DocumentIndexingPolicy.class);
+        final String[] defaultExcludePath = new String[]{Constants.DEFAULT_EXCLUDEDPATH};
```

Comment:
should not change the default behavoir of indexingpolicy, leave the includedpath null.